### PR TITLE
feat: billing tab for platform

### PR DESCRIPTION
**Pull request by Ryukemeister**
## What does this PR do?

- Adds billing tab for platform user to upgrade or downgrade their plans, update their stripe billing info and collect invoices

https://github.com/user-attachments/assets/aefe86b9-238d-4b71-b82b-96778ae51316

https://github.com/user-attachments/assets/5dda1c5f-6cde-4129-9278-3f13ccb1143e

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
This can be tested in the web app, go to `/settings/platform` and click the billing tab and try to upgrade or downgrade your plans or update billing info.

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]

**Comment by ThyMinimalDev** [comment]
checking that everything is working before merge

File: apps/api/v2/src/modules/billing/services/billing.service.ts [hunk 1]
```diff
@@ -43,12 +49,9 @@ export class BillingService implements OnModuleDestroy {
 
   async createSubscriptionForTeam(teamId: number, plan: PlatformPlan) {
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/billing/services/billing.service.ts:50]: in follow up pr, let's refactor this in 2 different functions doing only one thing

- createTeamBilling, which just returns the customerId or the whole platforBillingEntity
- redirectToSubscribeCheckout, which returns the url

File: apps/api/v2/src/modules/billing/controllers/billing.controller.ts [hunk 2]
```diff
@@ -69,13 +68,32 @@ export class BillingController {
     @Param("teamId") teamId: number,
     @Body() input: SubscribeToPlanInput
   ): Promise<ApiResponse<SubscribeTeamToBillingResponseDto | undefined>> {
-    const { status } = await this.billingService.getBillingData(teamId);
+    const { action, url } = await this.billingService.createSubscriptionForTeam(teamId, input.plan);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/billing/controllers/billing.controller.ts:72]: in a follow up pr call the two newly created functions one after the other

```
const customerId = await createTeamBilling
const url = await redirectToSubscribeCheckout(customerId)

...
```

**Review by ThyMinimalDev — APPROVED** [review]
LGTM, some refactor to do in follow up pr

**Pull request merged**